client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-email-notification/add-new-email-modal.tsx

```diff
@@ -49,36 +90,61 @@ export default function AddNewEmailModal( { toggleModal }: { toggleModal: () =>
 		<Modal
 			open={ true }
 			onRequestClose={ toggleModal }
-			title={ translate( 'Add your email address' ) }
+			title={ title }
 			className="notification-settings__modal"
 		>
-			<div className="notification-settings__sub-title">
-				{ translate( 'Please use only your number or one you have access to. ' ) }
-			</div>
+			<div className="notification-settings__sub-title">{ subTitle }</div>
 
 			<form className="configure-email-notification__form" onSubmit={ onSave }>
 				<FormFieldset>
 					<FormLabel htmlFor="name">{ translate( 'Name' ) }</FormLabel>
-					<FormTextInput id="name" name="name" disabled={ showCodeVerification } />
-					<div className="configure-email-notification__help-text">
-						{ translate( 'Give this email a nickname for your personal reference' ) }
-					</div>
+					<FormTextInput
+						id="name"
+						name="name"
+						value={ emailItem.name }
+						disabled={ showCodeVerification }
+						onChange={ ( event: { target: { value: string } } ) =>
+							setEmailItem( { ...emailItem, name: event.target.value } )
+						}
+					/>
+					{ ! isVerifyAction && (
+						<div className="configure-email-notification__help-text">
+							{ translate( 'Give this email a nickname for your personal reference' ) }
+						</div>
+					) }
 				</FormFieldset>
 
 				<FormFieldset>
 					<FormLabel htmlFor="email">{ translate( 'Email' ) }</FormLabel>
-					<FormTextInput id="email" name="email" disabled={ showCodeVerification } />
-					<div className="configure-email-notification__help-text">
-						{ translate( 'We’ll send a code to verify your email address.' ) }
-					</div>
+					<FormTextInput
+						id="email"
+						name="email"
+						value={ emailItem.email }
+						disabled={ showCodeVerification }
+						onChange={ ( event: { target: { value: string } } ) =>
+							setEmailItem( { ...emailItem, email: event.target.value } )
+						}
+					/>
+					{ ! isVerifyAction && (
+						<div className="configure-email-notification__help-text">
+							{ translate( 'We’ll send a code to verify your email address.' ) }
+						</div>
+					) }
 				</FormFieldset>
 
 				{ showCodeVerification && (
 					<FormFieldset>
 						<FormLabel htmlFor="code">
 							{ translate( 'Please enter the code you received via email' ) }
 						</FormLabel>
-						<FormTextInput id="code" name="code" />
+						<FormTextInput
+							id="code"
+							name="code"
+							value={ emailItem.code }
+							onChange={ ( event: { target: { value: string } } ) =>
+								setEmailItem( { ...emailItem, code: event.target.value } )
```

Comment:
Since we depend on the previous state value, we can pass a setter callback here as a good practice.  While unlikely we update multiple fields simultaneously so this is a non-blocker.

```
setEmailItem((prevState) => ({ ...prevState, code: event.target.value }));
```